src/api/app/controllers/webui/project_controller.rb

```diff
@@ -709,8 +709,8 @@ def add_maintained_project
 
   def remove_maintained_project
     authorize @project, :update?
-    @project.maintained_projects.delete(MaintainedProject.find_by(project: @maintained_project))
-    if @project.save
+    maintained_project = MaintainedProject.find_by(project: @maintained_project)
+    if maintained_project && @project.maintained_projects.destroy(maintained_project)
```

Comment:
Unless destroy gives any return value that would indicate if deleting it succeeded or failed.